config/routes.rb

```diff
@@ -19,5 +19,9 @@
     end
   end
 
+  namespace :api do
+    resources :service_providers, only: [:index]
```

Comment:
```diff
diff --git a/config/routes.rb b/config/routes.rb
index 8f281f1..104f400 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -14,14 +14,16 @@ Rails.application.routes.draw do
     get 'timeout' => 'users/sessions#timeout'
   end
 
-  namespace :users do
-    resources :service_providers do
-    end
-  end
+  get '/users/service_providers' => 'users/service_providers#index'
+  post '/users/service_providers' => 'users/service_providers#create'
+  get '/users/service_providers/new' => 'users/service_providers#new'
+  get '/users/service_providers/:id/edit' => 'users/service_providers#edit'
+  get '/users/service_providers/:id' => 'users/service_providers#show'
+  patch '/users/service_providers/:id' => 'users/service_providers#update'
+  put '/users/service_providers/:id' => 'users/service_providers#update'
+  delete '/users/service_providers/:id' => 'users/service_providers#destroy'
 
-  namespace :api do
-    resources :service_providers, only: [:index]
-  end
+  get '/api/service_providers' => 'api/service_providers#index'
 
   root to: 'home#index'
 end
```

kinda like the terse rails magic, but respect to the explicit style. what do you think?